### PR TITLE
fix compatibily trap between NC20 and NC21 #1811

### LINKED ISSUE #1811: An unhandled exception has been thrown during update V2.0.4 -> V2.0.5
## What is going wrong? What did you observe?
**Describe the bug**

Similar to #1770. Log below.

**Steps to reproduce the behavior**

Updated from web, then received error, and updated from occ command.

Also removed app via occ command line and attempted fresh install with occ command and received same error.

**Add a screenshot of the misbehavior**

N/A

## How should it work?
**What you expected to happen?**

Clean update

## Information about your polls installation
**Polls version:** 2.0.4 -> 2.0.5

**Fresh installation or update from a prior version (from which one)?**

- [x] First time install
- [x] Update from a prior version
  Version number of the previous version: 2.0.4 -> 2.0.5

**How did you install this version?(Appstore or describe installation)**

- [x] Installed from the appstore
- [x] Installed via occ
- [ ] Installed via extracting downloaded package to the apps folder

## Information about your Instance of Nextcloud
**Nextcloud version:** 21.0.3

### Server configuration (decide, if you think it is helpful)

**Database:**
- [ ] MySql version: 
- [x] MariaDB version: 10.3.29
- [ ] PostgreSQL version:
- [ ] Oracle version:
- [ ] SQLite:
- [ ] Other (add name and version):

**PHP version:**
- [ ] 7.3 or lower
- [x] 7.4
- [ ] 8.0

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

No

## Logs

````
An unhandled exception has been thrown:
ArgumentCountError: Too few arguments to function OCA\Polls\Migration\FixVotes::__construct(), 0 passed in /var/www/lib/private/Repair.php on line 129 and exactly 4 expected in /var/www/apps/polls/lib/Migration/FixVotes.php:50
Stack trace:
#0 /var/www/lib/private/Repair.php(129): OCA\Polls\Migration\FixVotes->__construct()
#1 /var/www/lib/private/legacy/OC_App.php(1043): OC\Repair->addStep()
#2 /var/www/lib/private/Installer.php(145): OC_App::executeRepairSteps()
#3 /var/www/core/Command/App/Install.php(68): OC\Installer->installApp()
#4 /var/www/apps/bookmarks/vendor/symfony/console/Command/Command.php(255): OC\Core\Command\App\Install->execute()
#5 /var/www/apps/bookmarks/vendor/symfony/console/Application.php(1009): Symfony\Component\Console\Command\Command->run()
#6 /var/www/apps/bookmarks/vendor/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
#7 /var/www/apps/bookmarks/vendor/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun()
#8 /var/www/lib/private/Console/Application.php(215): Symfony\Component\Console\Application->run()
#9 /var/www/console.php(100): OC\Console\Application->run()
#10 /var/www/occ(11): require_once('/var/www/...')
ubuntu@raspberry-pi:/var/www$ 
````

### PR DESCRIPTION
**Pull request by dartcafe**
fix #1811

### CODE REVIEW

File: lib/Migration/FixVotes.php [hunk 1]
```diff
@@ -72,20 +65,19 @@ public function getName() {
...
-					$this->voteMapper->fixVoteOptionText(
-						$option->getPollId(),
-						$option->getId(),
-						$option->getPollOptionTextStart(),
-						$option->getPollOptionText()
-					);
-				}
+		try {
+			$foundOptions = $this->optionMapper->findOptionsWithDuration();
+			\OC::$server->getLogger()->error(json_encode($foundOptions));
```
Comment (ChristophWurst) [inline lib/Migration/FixVotes.php:70]: this will now log an error for everyone, even for successful runs

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
I don't think the try-catch is necessary. What threw an exception was the borked DI resolution, not the actual method code

File: lib/Migration/FixVotes.php (hunk 1 above)
Comment (dartcafe) [reply lib/Migration/FixVotes.php]: 😆 Just removed it. Thanks. 👍

**Comment by dartcafe** [comment]
> 
> 
> I don't think the try-catch is necessary. What threw an exception was the borked DI resolution, not the actual method code

As I we run it as a repair step, an exception is thrown if the actual db scheme is not in the current status. The code before checked, if the db and the column exist, but break because of the dependency injection. From the version 3.x this all is history.

**Review by ChristophWurst — APPROVED** [review]

**Comment by dartcafe** [comment]
Thanks for merging. But I was wondering, why the origin branch was not found. 😄

**Pull request merged**